src/pe/header.rs

```diff
@@ -814,20 +831,21 @@ impl Header {
         Ok(Header {
             dos_header,
             dos_stub,
+            rich_header,
             signature,
             coff_header,
             optional_header,
         })
     }
 }
 
-impl ctx::TryIntoCtx<scroll::Endian> for Header {
+impl<'a> ctx::TryIntoCtx<scroll::Endian> for Header<'a> {
     type Error = error::Error;
 
     fn try_into_ctx(self, bytes: &mut [u8], ctx: scroll::Endian) -> Result<usize, Self::Error> {
         let offset = &mut 0;
         bytes.gwrite_with(self.dos_header, offset, ctx)?;
-        bytes.gwrite_with(self.dos_stub, offset, ctx)?;
+        bytes.gwrite_with(self.dos_stub, offset, ())?;
```

Comment:
i'd expect this to use the same ctx here but maybe there's a good reason

Reply:
Yeah.. since it's a code I wrote a while ago, and I now realized that `Copy` impl is the whole problem why I assigned the  explicit lifetime to the structs. Simply because [`std::vec::Vec<T>`](https://doc.rust-lang.org/std/vec/struct.Vec.html) doesn't implement `Copy`. So `RichHeader`, `RichMetadata` and the `Header` are entiely not be able to impl `Copy`. Maybe I need to remove `Vec<u8>` from those structs somehow

Reply:
Reopening. Yeah, the issue is that the `DosStub` is no longer a fixed size that the compiler knows at the compile time, and it should be dynamically sized. So `Pread` and `Pwrite` derives cannot be implemented. I guess there are no way to avoid using of `Vec<u8>`, but maybe in hacky way, like below, which is crazy (really) and we mustn't do that.

```rs
pub struct DosStub {
    pub data: [u8; 0x1000],
    pub real_len: usize,
}
```

I'm not faimilar with `scroll` or `Pread`, ``Pwrite` though, do you have any options?

```rs
error[E0308]: mismatched types
   --> src\pe\header.rs:887:45
    |
887 |         bytes.gwrite_with(dos_stub, offset, ctx)?;
    |               -----------                   ^^^ expected `()`, found `Endian`
    |               |
    |               arguments to this method are incorrect
```